File: golang/divergent-modifications/good/financial_calculator.go
package main

type financialCalculator struct{}

func NewFinancialCalculator() FinancialCalculator {
	return &financialCalculator{}
}

func (fc financialCalculator) CalculateInterest(principal, rate float64, time int) float64 {
	return principal * rate * float64(time)
}

func (fc financialCalculator) CalculateTax(income, deductions float64) float64 {
	taxableIncome := income - deductions
	if taxableIncome <= 50000 {
		return taxableIncome * 0.1
	} else if taxableIncome <= 100000 {
		return 5000 + (taxableIncome-50000)*0.2
	} else {
		return 15000 + (taxableIncome-100000)*0.3
	}
}

File: python/divergent-modifications/good/financial_calculator.py
class FinancialCalculator:
    def calculate_interest(self, principal: float, rate: float, time: int) -> float:
        return principal * rate * time

    def calculate_tax(self, income: float, deductions: float) -> float:
        taxable_income = income - deductions
        if taxable_income <= 50000:
            return taxable_income * 0.1
        elif taxable_income <= 100000:
            return 5000 + (taxable_income - 50000) * 0.2
        else:
            return 15000 + (taxable_income - 100000) * 0.3
